openretina/utils/model_utils.py

```diff
@@ -51,3 +52,33 @@ def on_validation_end(self, trainer, pl_module):
             trainer.optimizers = [new_optimizer]  # Replace the optimizer in the trainer
 
         self.prev_lr = current_lr
+
+
+def get_core_output_based_on_dimensions(model_config):
+    if len(model_config.in_shape) == 5:
+        input_shape = model_config.in_shape[1:]
+    else:
+        input_shape = model_config.in_shape
+    ch, t, h, w = input_shape[:]
+    layers = len(model_config.core.temporal_kernel_sizes)
+
+    for i in range(layers):
+        temp_kernel_size = model_config.core.temporal_kernel_sizes[i]
+        spatial_kernel_size = model_config.core.spatial_kernel_sizes[i]
+        if isinstance(spatial_kernel_size, int):
+            spatial_kernel_size = (spatial_kernel_size, spatial_kernel_size)
+        temporal_dilation = (
+            1 if "temporal_dilation" not in model_config.core.keys() else model_config.core.temporal_dilation
+        )
```

Comment:
if model_config.core is dict like, you can write:
```suggestion
        temporal_dilation = model_config.core.get("temporal_dilation", 1)
```